test/unit/queries/proxy_rule_query_test.rb

```diff
@@ -51,4 +52,16 @@ class ProxyRuleTest < ActiveSupport::TestCase
       assert_equal [], query.search_for('')
     end
   end
+
+  test '#search_for finds proxy_rule with special characters' do
+    ThinkingSphinx::Test.rt_run do
+      backend_api = FactoryBot.build_stubbed(:backend_api)
+      perform_enqueued_jobs(only: IndexProxyRuleWorker) do
+        proxy_rule = FactoryBot.create(:proxy_rule, owner: backend_api, pattern: '/*-+/.,!_')
```

Comment:
It would be nice to have multiple different tests for this, one for each special character:

```ruby
%w[/*-+/.,!_'].each do |char|
   pattern = "path#{char}"
   proxy_rule = FactoryBot.create(:proxy_rule, owner: backend_api, pattern: pattern)
   query      = ProxyRuleQuery.new(owner_type: proxy_rule.owner_type, owner_id: proxy_rule.owner_id)
   assert_equal [proxy_rule], query.search_for(pattern)
```

Reply:
@hallelujah I created multiple tests (one for each character) and moved them in a new file under /test/unit/indices, let me know if it makes more sense now, thanks.